### PR TITLE
Fix socket closing

### PR DESCRIPTION
**Pull request by take-cheeze**
Seems like it needs to call `closesocket` instead of `close` in Windows.
https://msdn.microsoft.com/en-us/library/windows/desktop/ms737582(v=vs.85).aspx

[`BasicSocket#close`](https://github.com/iij/mruby-socket/blob/ab54185005ec87fe4f5b10df95ad29659884141b/src/socket.c#L721) does support it but in mruby-io it isn't.

### CODE REVIEW

File: mrbgems/mruby-socket/test/unix.rb [hunk 1]
```diff
@@ -1,5 +1,3 @@
-unless SocketTest.win?
```
Comment (matz) [inline mrbgems/mruby-socket/test/unix.rb]: Is this change intentional? Does UNIXSocket test run OK for Windows?

Comment (take-cheeze) [reply mrbgems/mruby-socket/test/unix.rb]: At least many `mruby-io` related stuff is failing too so I don't think that line will do anything good.
(Unless you want to hide the CI test errors in windows.)
You should exclude whole `mruby-io` and `mruby-socket` mrbgem instead of doing those kind of thing.

This PR only makes fix to exceptions inside Windows' dynamic library, that's all.
I've fixed segv error and there is the rest errors.

Comment (mattn) [reply mrbgems/mruby-socket/test/unix.rb]: I suggest to merge this since CI is failing (crash) on Windows.

crash

**Comment by take-cheeze** [comment]
After some rethinking `BasicSocket#is_socket=` should be renamed to `_is_socket=` since it's for internal use.

**Comment by matz** [comment]
#3878 merged.
